tracer/src/Datadog.Trace/Ci/CiEnvironment/GitCommandGitInfoProvider.cs

```diff
@@ -0,0 +1,144 @@
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "ls-remote --get-url",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (repositoryOutput?.ExitCode == 0)
+            {
+                localGitInfo.Repository = repositoryOutput.Output.Trim();
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting repository URL: {repositoryOutput?.Error}");
+            }
+
+            // Get the branch name
+            var branchOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: "rev-parse --abbrev-ref HEAD",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (branchOutput?.ExitCode == 0 && branchOutput.Output.Trim() is { Length: > 0 } branchName && branchName != "HEAD")
+            {
+                localGitInfo.Branch = branchName;
+            }
+            else
+            {
+                localGitInfo.Errors.Add($"Error getting branch name: {branchOutput?.Error}");
+            }
+
+            // Get the remaining data from the log -1
+            var gitLogOutput = ProcessHelpers.RunCommand(
+                new ProcessHelpers.Command(
+                    cmd: "git",
+                    arguments: """log -1 --pretty='%H|,|%at|,|%an|,|%ae|,|%ct|,|%cn|,|%ce|,|%B'""",
+                    workingDirectory: gitDirectory.FullName,
+                    useWhereIsIfFileNotFound: true));
+            if (gitLogOutput?.ExitCode != 0)
+            {
+                localGitInfo.Errors.Add($"Error getting git log: {gitLogOutput?.Error}");
+                return false;
+            }
+
+            var gitLogDataArray = gitLogOutput.Output.Trim().Split(["|,|"], StringSplitOptions.None);
+            if (gitLogDataArray.Length < 8)
+            {
+                localGitInfo.Errors.Add($"Git log output does not contain the expected number of fields: {gitLogOutput.Output}");
+                return false;
+            }
+
+            // Parse author and committer dates from Unix timestamp
+            if (!long.TryParse(gitLogDataArray[1], out var authorUnixDate))
+            {
+                localGitInfo.Errors.Add("Error parsing author date from git log output");
+                return false;
+            }
+
+            if (!long.TryParse(gitLogDataArray[4], out var committerUnixDate))
+            {
+                localGitInfo.Errors.Add("Error parsing committer date from git log output");
+                return false;
+            }
+
+            // Populate the localGitData struct with the parsed information
+            localGitInfo.Commit = gitLogDataArray[0];
+            localGitInfo.AuthorDate = UnixTimeStampToDateTime(authorUnixDate);
```

Comment:
```suggestion
            localGitInfo.AuthorDate = DateTimeOffset.FromUnixTimeSeconds(authorUnixDate);
```

Reply:
🤦🏻 done in [6b0b8a9](https://github.com/DataDog/dd-trace-dotnet/pull/6259/commits/6b0b8a9ad9f70add2b7296b03b29ea484e47a859)